kernel/core/io/file.cpp

```diff
@@ -4117,6 +4147,35 @@ namespace NumeRe
         _styles.Parse(sStylesContent.c_str());
         tinyxml2::XMLElement* cellXfs = _styles.FirstChildElement()->FirstChildElement("cellXfs");
 
+        // Initialize Set with Standard Excel Datetime IDs: 18..21, 45..47
+        std::set<int> vStandardDateTimeIds = {14, 15, 16, 17, 18, 19, 20, 21, 22, 45, 46, 47};

```

Comment:
As said before, I recommend to have separate sets for time and date values (14-17 are date, the others are time values)